KEP for CoreDNS

rajansandeep opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
- If this is your first contribution, read our Getting Started guide https://github.com/kubernetes/community#your-first-contribution
- If you are editing SIG information, please follow these instructions: https://git.k8s.io/community/generator
  You will need to follow these steps:
  1. Edit sigs.yaml with your change 
  2. Generate docs with `make generate`. To build docs for one sig, run `make WHAT=sig-apps generate`
-->

janetkuo committed:
Update kubectl rollout proposal to reflect current implementation

mikedanese committed:
proposal: bound service account tokens

Proposal to improve the utility of service account tokens by supporting
audience, time and key binding.

@kubernetes/sig-auth-api-reviews

rajansandeep committed:
initial commit

KEP coredns for GA only

rajansandeep commented:
@chrisohaver @fturib @johnbelamaric PTAL.

fturib commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+### Non-Goals
+
+* Translation of CoreDNS configmap back to kube-dns.
+* Retaining the configmap of a custom/modified CoreDNS image of CoreDNS to CoreDNS.
```
review wording "CoreDNS image of CoreDNS to CoreDNS"

fturib commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+* Translation of CoreDNS configmap back to kube-dns.
+* Retaining the configmap of a custom/modified CoreDNS image of CoreDNS to CoreDNS.
+* Translation of out-of configmap configuration of kube-dns.
+* Documentation on "How-to switch to CoreDNS as the default DNS service" for maintainers of other tools.
```
the How-to may not be neither a Goal neither a non-Goal.
We may just give for info, or do not talk at all.

fturib commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+* kubeadm
+* kube-up
+* minikube
+* kops
```
should we talk about "kops" here ? 
the versions are delayed - and current PR is still waiting.

fturib commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+## Motivation
+
+* CoreDNS is a CNCF project which is in incubation status. It is dedicated to DNS Service and includes several Kubernetes related enhancement that place it as same functionality of kube-dns regarding Kubernetes usage (Federation, Autopath, Stubdomains, etc.).
```
need to check with Chris. But I though Autopath is not a function of kube-dns. Federation ?

chrisohaver replied:
Autopath is not a kube-dns feature.  It's unique to CoreDNS.  Should be added to the section later...

I think this sentence is saying "CoreDNS can provide kube-dns functions that are not in the k8s DNS spec, such as Federation, and Stubdomains."
You can lead into the next section with, "CoreDNS has some key advantages vs kube-dns..."

fturib reviewed (commented):
Great doc.
Lets sync-up tomorrow on the comments.
Would like to have this doc proposed as PR in k8s/community for tomorrow COB

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+## Summary
+
+CoreDNS is a CNCF project dedicated to the DNS service. It has developed a K8s interface that is conform to existing kube-dns specs.
```
"that is conform" -> "that conforms"

Also to be clear add, "... that conforms to kube-dns specs, _and therefore can replace kube-dns as the cluster dns for service discovery_."
Also link "specs" to the specs here.
Also, I dont think there are called "kube-dns specs" - i think they are the "Cluster DNS Service Discovery Specs" or something similar. Should check that.  You can get this from the Kubernetes plugin README.

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+### Goals
+
+* Have CoreDNS run as the default DNS server either after a new installation or an upgrade of the existing installation.
```
I think foremost, we want the Kubernetes project to adopt CoreDNS as the "standard" k8s Cluster Service Discovery DNS, replacing kube-dns.  It then follows that upgrade tools, such as kubeadm, etc should deploy CoreDNS instead of kube-dns.

But, it's unclear to me if Kubernetes has a "standard" like this ... they may need to create one.  Or perhaps the standard is "what kubeadm does"

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+## Motivation
+
+* CoreDNS is a CNCF project which is in incubation status. It is dedicated to DNS Service and includes several Kubernetes related enhancement that place it as same functionality of kube-dns regarding Kubernetes usage (Federation, Autopath, Stubdomains, etc.).
+* CoreDNS has fewer moving parts than kube-dns, taking advantage of the caddy style plugin making it a single executable and single process.
```
I think the biggest advantage is that it is easily extensible and maintainable using a plugin architecture.  Should be a bullet here.

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+### Goals
+
+* Have CoreDNS run as the default DNS server either after a new installation or an upgrade of the existing installation.
+* Configmap of current kube-dns would be handled by being translated into the corresponding configuration of CoreDNS.
```
I think this bullet and the last bullet can be combined into:
   * Provide a kube-dns to CoreDNS upgrade path with configuration translation.

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+* Translation of CoreDNS configmap back to kube-dns.
+* Retaining the configmap of a custom/modified CoreDNS image of CoreDNS to CoreDNS.
+* Translation of out-of configmap configuration of kube-dns.
```
might need to explain this more... or perhaps not include here and instead explain in the proposal.

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+* Passing all e2e conformance tests and DNS related.
+* Process in place for getting CoreDNS image in gcr.io.
+* CoreDNS be installed as the default DNS service.
```
bullets 2 and 3 are goals...  do they belong in graduation criteria?  Seems redundant.

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+Currently, we are following the road-map defined [here](https://github.com/kubernetes/features/issues/427). CoreDNS is Beta in Kubernetes v1.10, which can be installed as an alternate to kube-dns.
+The purpose of this proposal is to graduate CoreDNS to GA and replace the current implementation of kube-dns service by CoreDNS.
+
+## Motivation
```
I think the things we say in "Motivation" should be things that fit with "We want to do this because..."
e.g. We want the Cluster DNS to be more flexible and extensible than what kube-dns currently offers.

IMO that is the core motivation.  Addtional benefits are, that it happens to perform better (but if it didnt we'd still be trying to convince them to move to CoreDNS, so its not really a core motivation)

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+* kops
+
+For other tools, each maintainer would have to add the upgrade to CoreDNS.
+It is not a mandatory one shot operation, as for at least some time, kube-dns and CoreDNS will provide an equivalent service.
```
Not sure what you mean here. Are you saying that its not mandatory that all kubernetes deployment tools in existence would provide an upgrade path to coredns at the same time?

chrisohaver commented on keps/sig-network/0006-20180314-coredns-GA-proposal.md:
```diff
@@ -0,0 +1,134 @@
+
+The following can be expected when CoreDNS is graduated to GA.
+
+#### Kubeadm
```
I think we should lead with CoreDNS "use cases" here.  Otherwise the proposal comes across as very deployment and upgrade centric.

chrisohaver reviewed (changes requested):
There seems to be a bit of repetition... we may be able to prune some sections down - or perhaps reword.
We should avoid the "passive voice" in proposals to curb ambiguity.

jessfraz committed:
add ProcMount option

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

Commit:
Move provider Cloud Controller Manager repo reqs to kep

Moves the recently approved provider Cloud Controller Manager repository
requirements to the Cloud Controller Manager kep. Also integrates
corrections and clarifications from WG-Cloud-Provider.

mikedanese committed:
design: improved service account token volumes

rramkumar1 committed:
Add a retroactive KEP for IPVS

Commit:
simplify the cherry pick process documentation

The cherry-pick instruction guide was quite out of date.  This
commit dramatically simplifies the doc.  Also, based on watching a
new user attempt a cherry pick (a common occurance around release
time with the revolving staffing of the release team), a few
clarification and setup issues are addressed around the
hack/cherry_pick_pull.sh script pre-requisites.

This is an interim fix as the doc will need updated again to capture
the pending prow based method when that becomes preferred and the
file should be merged with the various other branch and release
maintenance documents that are out there.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

k8s-ci-robot committed:
Merge pull request #2015 from mahuihuang/fix_grammar_mistake

Fix grammar mistake

CaoShuFeng committed:
fix AdmissionConfiguration yaml format

vijaymateti committed:
Fix for duplicate title on Contributors guide README

k8s-ci-robot committed:
Merge pull request #2004 from ConnorDoyle/wg-ml-organizers

Append to ML WG organizers.

castrojo committed:
Remove afternoon docs session

k8s-ci-robot committed:
Merge pull request #2016 from CaoShuFeng/pod-priority-resourcequota

fix AdmissionConfiguration yaml format

castrojo committed:
Spelling fixes to pass bot, add more information on google groups

castrojo committed:
Add sig lead list reference, update slack admin information

k8s-ci-robot committed:
Merge pull request #2021 from castrojo/add-lead-list

Add sig lead list reference, update slack admin information

k8s-ci-robot committed:
Merge pull request #1986 from kimsterv/patch-1

Include more instructions on WG creation.

glikson committed:
typo in scheduler alg doc

k8s-ci-robot committed:
Merge pull request #2020 from castrojo/update-schedule

Remove afternoon docs session

k8s-ci-robot committed:
Merge pull request #2019 from vijaymateti/master

Fix for duplicate title on Contributors guide README

k8s-ci-robot committed:
Merge pull request #2007 from castrojo/add-code-site

Add code search and contributor guide, remove reviewable

Commit:
update issue triage doc url

The issue triage doc has moved from:
	contributors/devel/issues.md
to:
	contributors/guide/issue-triage.md

Signed-off-by: Tim Pepper <tpepper@vmware.com>

k8s-ci-robot committed:
Merge pull request #2024 from tpepper/contrib_guide

update issue triage doc url

mbohlool committed:
CRD versioning design proposal

k8s-ci-robot committed:
Merge pull request #1991 from freehan/pod-ready

add KEP for pod ready++

marpaia committed:
Move the doc to new location and remove scratch section

marpaia committed:
Copy of Briant Grant's Kubernetes Resource Management

dghubble committed:
Archive SIG on-premise, now a slack channel

k8s-ci-robot committed:
Merge pull request #2027 from marpaia/resource-management

Add Briant Grant's Design Proposal on Kubernetes Resource Management

grantbow committed:
renamed sig-contributor-experience google group

As per
https://groups.google.com/d/topic/kubernetes-sig-contribex/sFWBwGUHszc/discussion
the "kubernetes-wg-contribex" group name is now "kubernetes-sig-contribex".

k8s-ci-robot committed:
Merge pull request #2023 from glikson/patch-1

typo in scheduler alg doc

linyouchong committed:
Update error in e2e-tests.md

`go run hack/e2e.go --help` should be `go run hack/e2e.go -- --help`

k8s-ci-robot committed:
Merge pull request #2036 from grantbow/master

renamed sig-contributor-experience google group

Commit:
add docs for The 1:1 Hour

Commit:
add docs for mentor guidelines.md

k8s-ci-robot committed:
Merge pull request #2025 from dghubble/master

Remove SIG on-premise since its now a slack channel

fatsheep9146 committed:
Fix mistakes about the example of resource account for burstable pods and typo mistake

k8s-ci-robot committed:
Merge pull request #2046 from fatsheep9146/fix-mistakes

Fix mistakes about the example of resource account for burstable pods and typo mistake

johnarok committed:
Add note on KUBE_TIMEOUT to resolve timeout panic

k8s-ci-robot committed:
Merge pull request #2039 from linyouchong/patch-1

Fix error in e2e-tests.md

johnarok committed:
Formatting changes

k8s-ci-robot committed:
Merge pull request #2047 from johnarok/johnarok-patch-testingtimeoutinfo

Add note on KUBE_TIMEOUT to resolve timeout panic

spzala committed:
Describe new labels in triage guidelines

Recently we have added new labels to identify issues that are candidates for close.
These issues are typically the one which are not goint to close via a fix. Add
description in triage guidelines about when to use these labels.
Related PR is,
kubernetes/test-infra#7554

castrojo committed:
Add links to other OSS communities using Discourse

spzala committed:
Add SIG IBMCloud

A doc page for the newly created sig-ibmcloud.

pmorie committed:
Add rules for donating repositories to kubernetes-sigs

k8s-ci-robot committed:
Merge pull request #1987 from marpaia/declarative-application-management

Copying Brian Grant's article on declarative application management from Drive

k8s-ci-robot committed:
Merge pull request #2049 from pmorie/donating-repos-to-kubernetes-sigs

Add rules for donating repositories to kubernetes-sigs

k8s-ci-robot committed:
Merge pull request #2048 from spzala/sigibmcloud

Add SIG IBMCloud

Commit:
docs for mentoring activities at events

k8s-ci-robot committed:
Merge pull request #2044 from parispittman/mentorguide

add docs for mentorguidelines.md

k8s-ci-robot committed:
Merge pull request #2041 from parispittman/the11hour

add docs for The 1:1 Hour

Commit:
Adding chair for SIG-Windows

k8s-ci-robot committed:
Merge pull request #1970 from spzala/triagelabels

Describe new labels in triage guidelines

k8s-ci-robot committed:
Merge pull request #290 from janetkuo/update-deployment-proposal

Update kubectl rollout proposal to reflect current implementation

justaugustus committed:
Update sigs.yaml to incorporate SIG Azure changes

k8s-ci-robot committed:
Merge pull request #2056 from PatrickLang/sigwin-update

Adding chair for SIG-Windows

k8s-ci-robot committed:
Merge pull request #1460 from mikedanese/svcacct-jwts

proposal: bound service account tokens

shubheksha committed:
Update issue link for sig/storage

Current link points to issues + PRs

castrojo committed:
Be more explicit about the graduation criteria

castrojo committed:
Remove myself as graph-o-the-week maintainer

justaugustus committed:
sig-azure: Update autogenerated files

castrojo committed:
Add moderation policy requirement

k8s-ci-robot committed:
Merge pull request #1980 from tpepper/master

clarify some parts of the cherry pick process

k8s-ci-robot committed:
Merge pull request #2011 from castrojo/community-forum

KEP-0007 - Community forum

G-Harmon committed:
update aggregated-api-servers.md: fix a typo

k8s-ci-robot committed:
Merge pull request #2054 from mbohlool/crd-versioning-noconv

CRD versioning design proposal [No Conversion]

k8s-ci-robot committed:
Merge pull request #2059 from castrojo/jorge-doesnt-devstat

Remove myself as graph-o-the-week maintainer

k8s-ci-robot committed:
Merge pull request #2053 from parispittman/eventsmentoring

docs for mentoring activities at events

k8s-ci-robot committed:
Merge pull request #2057 from shubheksha/sig/storage-fix-issue-link

Update issue link for sig/storage

guineveresaenger committed:
Adds new contributor playground

This folder is meant as a location for participants of the new contributor track at KubeCon Copenhagen 2018 to safely try out submitting pull requests and familiarizing themselves with the workflow.

vikaschoudhary16 committed:
Add proposal for node-level User Namespaces remapping support

castrojo committed:
Initial commit

venezia committed:
Updating wg-apply information with video conference URL

venezia committed:
Updating wg-apply information with video conference URL

pmorie committed:
Fix typo in governance requirements doc

castrojo committed:
Add reviewer fixes

Reword bullets to have a consistant voice. Add link to Slack guidelines.
Add link to k8s community values. Clarify that this policy applues
to all official mediums.

k8s-ci-robot committed:
Merge pull request #2073 from pmorie/governance-requirements-typo

Fix typo in governance requirements doc

chris-short committed:
projects.md: Typo

Correction to GH handle for @chris-short (I know, it's confusing for me too)

Signed-off-by: Chris Short <chris@chrisshort.net>

k8s-ci-robot committed:
Merge pull request #2075 from chris-short/chris-short-001

projects.md: Typo

ConnorDoyle committed:
Fix typo in wg-machine-learning readme.

k8s-ci-robot committed:
Merge pull request #2076 from ConnorDoyle/ml-typo

Fix typo in wg-machine-learning readme.

k8s-ci-robot committed:
Merge pull request #2070 from venezia/update-wg-apply

Update wg apply information with video conference URL

pwittrock committed:
Remove friction from SIG Charter process through better documentation.

castrojo committed:
Add link to slack channel

k8s-ci-robot committed:
Merge pull request #2079 from castrojo/add-slack

Add link to slack channel

zparnold committed:
Remove contributor page

now that K8s 1.10 has been released

zparnold committed:
update to remove 1.10 outdated links

jberkus committed:
Added details for New Contributor Summit.

k82cn committed:
Doc about PID pressure taint.

Signed-off-by: Da K. Ma <klaus1982.cn@gmail.com>

wojtek-t committed:
KEP: efficient node heartbeats

shubheksha committed:
Fix broken links

jberkus committed:
Fixed links for NCW per comments.

k8s-ci-robot committed:
Merge pull request #2082 from jberkus/new_contrib_workshop_detail

Added details for New Contributor Summit.

k8s-ci-robot committed:
Merge pull request #2087 from shubheksha/flexvolume-fix-links

Fix broken links in flexvolume guide

guineveresaenger committed:
changed label to area label

AdamDang committed:
Typo fix: we wil->we will

we wil->we will

k8s-ci-robot committed:
Merge pull request #2063 from G-Harmon/patch-1

update aggregated-api-servers.md: fix a typo

k8s-ci-robot committed:
Merge pull request #2080 from zparnold/remove-contrib-page

Remove contributor page

Commit:
adding current contributor track info

Commit:
adding current contrib track info

k8s-ci-robot committed:
Merge pull request #2092 from parispittman/currcontribtrack

adding info for current contrib track

ingvagabund committed:
KEP: Promote sysctl annotations to fields

AdamDang committed:
Typo fix: metada->metadata

metada->metadata

k8s-ci-robot committed:
Merge pull request #2094 from AdamDang/patch-5

Typo fix: metada->metadata

k8s-ci-robot committed:
Merge pull request #2066 from guineveresaenger/kubecon-practice-folder

Adds new contributor playground [WIP]

k8s-ci-robot committed:
Merge pull request #2061 from justaugustus/azure-lteam

Update SIG Azure org structure

Commit:
Fix Greg's name

k8s-ci-robot committed:
Merge pull request #2095 from evankanderson/master

Fix Greg's name

Commit:
add info on joining Google Groups

As evidenced by a post to kubernetes-dev asking to be added to the list,
the current info on mailing lists may not be completely clear to
readers.  This commit makes the wording more explicit and includes a
link to Google Groups' info on how to find and join a group.

A broken link to the sig-list.md is also corrected.

Signed-off-by: Tim Pepper <tpepper@vmware.com>

k8s-ci-robot committed:
Merge pull request #2096 from tpepper/master

add info on joining Google Groups

cblecker committed:
Set pruning settings for dep

cblecker committed:
Update deps

guintestdummy committed:
fixing area label

k8s-ci-robot committed:
Merge pull request #2098 from guintestdummy/areafix

fixing area label

castrojo committed:
Update the room for the welcome introduction

k8s-ci-robot committed:
Merge pull request #2099 from castrojo/new-room

Update the room for the welcome introduction

k8s-ci-robot committed:
Merge pull request #2097 from cblecker/dep-update

Update deps

guineveresaenger committed:
This is a practice commit

This is a paragraph describing your intent and the background for this pull request

jonasrosland committed:
Adding a note from KubeCon + CloudNativeCon in Copenhagen

Welcoming all new contributors

Commit:
Added link to kubernetes/community

When navigating the readme you can now click to be redirected to the repository

mitsutaka committed:
Add first practice commit

guineveresaenger committed:
Addresses review: text is in two lines

For better reading, text has been separated more clearly.

jonasrosland committed:
Added a proper title to the Markdown file

Looks better this way

k8s-ci-robot committed:
Merge pull request #2103 from guineveresaenger/kubecon

This is a practice commit

mitsutaka committed:
Resolve conflict

k8s-ci-robot committed:
Merge pull request #2109 from jonasrosland/kubecon

Adding a note from KubeCon + CloudNativeCon in Copenhagen

k8s-ci-robot committed:
Merge pull request #2089 from AdamDang/patch-4

Typo fix: we wil->we will

Commit:
Add New contributor summit notes

I took some notes during the New contributor summit notes.

Here they are

k8s-ci-robot committed:
Merge pull request #1639 from nitkon/patch-1

Update e2e-tests.md

Commit:
Fix typo and comment

Signed-off-by: Julien Garcia Gonzalez <garciagonzalez.julien@gmail.com>

k8s-ci-robot committed:
Merge pull request #2119 from jgsqware/contributor-notes

Add New contributor summit notes

Commit:
solve typo error

k8s-ci-robot committed:
Merge pull request #2120 from angao/check

Solve typo error

Commit:
Word error

AdamDang committed:
Typo fix: make is easy->make it easy/initilization->initiliazation

Fix two typos:
make is easy->make it easy
initilization->initiliazation

warmchang committed:
Add the minimum go version for release 1.11+

Signed-off-by: William Zhang <warmchang@outlook.com>

k8s-ci-robot committed:
Merge pull request #2124 from AdamDang/patch-5

Typo fix: make is easy->make it easy/initilization->initialization

k8s-ci-robot committed:
Merge pull request #2123 from angao/check

Word error

liubin committed:
Link to new target

AdamDang committed:
[cri-windows.md] Typo fix: configration->configuration

configration->configuration

k8s-ci-robot committed:
Merge pull request #2129 from AdamDang/patch-6

[cri-windows.md] Typo fix: configration->configuration

liztio committed:
KEP for kubeadm masterconfiguration beta

k8s-ci-robot committed:
Merge pull request #2111 from mitsutaka/kubecon

Add first practice commit

k8s-ci-robot committed:
Merge pull request #2107 from gitirabassi/kubecon/gitirabasi

Added link to kubernetes/community

pwittrock committed:
SIG CLI KEP for Kustomize subproject

k8s-ci-robot committed:
Merge pull request #2078 from pwittrock/charter

Remove friction from SIG Charter process through better documentation.

AdamDang committed:
Typo fix: Kuberntes->Kubernetes

Kuberntes->Kubernetes

AdamDang committed:
Update application_service_definition_notes.md

Commit:
Typo error: repeat 'from'

gerl1ng committed:
Add China to the list of events

KubeCon + CloudNativeCon is now held thrice a year.
China is added to the list of events.

Signed-off-by: Marius Gerling <marius@gerl1ng.de>

k8s-ci-robot committed:
Merge pull request #2127 from liubin/patch-1

Link to new target

k8s-ci-robot committed:
Merge pull request #2134 from angao/scalability

Typo error: repeat 'from'

Commit:
Typo fix: run make regenerate the files, focussed=>focused

AdamDang committed:
Typo fix in commands: kubeclt->kubectl

kubeclt->kubectl

coopr committed:
Link to Playlist view of Youtube videos

The previous version of the Youtube link auto-played the most recent video, and tucked the historical recordings off to the side.

This version of the Youtube link focuses on the historical list of recordings, and avoids potentially intrusive auto-play.

k8s-ci-robot committed:
Merge pull request #2140 from AdamDang/patch-8

Typo fix in commands: kubeclt->kubectl

castrojo committed:
Explicitly list the steering committee as the final escalation point

k8s-ci-robot committed:
Merge pull request #2141 from coopr/patch-1

Link to Playlist view of Youtube videos

spiffxp committed:
Update company

k8s-ci-robot committed:
Merge pull request #2143 from spiffxp/update-company

Update company

kindule committed:
Grammatical errors fix

k8s-ci-robot committed:
Merge pull request #2138 from angao/multicluster-1

Typo fix: run make regenerate the files, focussed=>focused

austbot committed:
Adds WHAT env variable description
When contributing specific parts of k8s, its helpful to know how to slice the build to only build a certain binary.

k8s-ci-robot committed:
Merge pull request #2132 from pwittrock/kustomize

SIG CLI KEP for Kustomize subproject

k8s-ci-robot committed:
Merge pull request #2144 from kindule/master

Grammatical errors fix of the container-storage-interface.md

k8s-ci-robot committed:
Merge pull request #2142 from austbot/update-build-docs

Adds WHAT env variable description

ravisantoshgudimetla committed:
Remove confusing aggregator list comment

k8s-ci-robot committed:
Merge pull request #2145 from ravisantoshgudimetla/master

Remove confusing aggregator list comment

austbot committed:
Corrects location of core defaults.go based on current kubernetes/kubernetes repo.

msau42 committed:
Topology aware dynamic provisioning design

Commit:
adding contributor summit notes

k8s-ci-robot committed:
Merge pull request #2136 from gerl1ng/EventChina

Add China to the list of events

k8s-ci-robot committed:
Merge pull request #2150 from parispittman/summitnotes

adding contributor summit EU notes

guineveresaenger committed:
dd videos to Contributor Workshop outline

The New Contributor Workshop at KubeConEU was well attended and captured on vide
o. This commit adds the video playlist to the course outline.

guineveresaenger committed:
Adds link to outline and video to README in new contributor playground folder.

k8s-ci-robot committed:
Merge pull request #2093 from ingvagabund/promote-sysctl-annotations-kep

KEP: Promote sysctl annotations to fields

k8s-ci-robot committed:
Merge pull request #2067 from vikaschoudhary16/userns

Add proposal for node-level User Namespaces remapping support

k8s-ci-robot committed:
Merge pull request #2152 from guineveresaenger/workshopvideo

Workshop video

shubheksha committed:
sig-storage: add kubecon eu 2018 talks to resources

wojtek-t committed:
Remove doc that is elsewhere now

k8s-ci-robot committed:
Merge pull request #2131 from liztio/kubeadm-config-beta

KEP for kubeadm masterconfiguration beta

k8s-ci-robot committed:
Merge pull request #2153 from wojtek-t/remove_performance_tool

Remove doc that is in perf-tests repo now

k8s-ci-robot committed:
Merge pull request #1942 from hogepodge/cloud-controller-manager

Move provider Cloud Controller Manager repo requirements to KEP

MHBauer committed:
remove extra parenthesis

k8s-ci-robot committed:
Merge pull request #2154 from MHBauer/patch-1

remove extra parenthesis

k8s-ci-robot committed:
Merge pull request #1973 from mikedanese/sats

design: improved service account token volumes

castrojo committed:
Correct the timezone links to be more generic

AdamDang committed:
Typo fix: excuted->executed

Fix some typos.

k8s-ci-robot committed:
Merge pull request #2155 from castrojo/timezones-argh

Correct the timezone links to be more generic

k8s-ci-robot committed:
Merge pull request #2156 from AdamDang/patch-9

Typo fix: excuted->executed

k8s-ci-robot committed:
Merge pull request #2125 from warmchang/go-version

Add the minimum go version for release 1.11+

Commit:
add SIG VMware ML and cloud-provider-vsphere subproject

ant31 committed:
Add issues links

k8s-ci-robot committed:
Merge pull request #2160 from ant31/patch-1

Add issues links

k8s-ci-robot committed:
Merge pull request #2069 from castrojo/moderation

Add community moderation guidelines

castrojo committed:
Correct UTC time for meeting

k8s-ci-robot committed:
Merge pull request #2161 from castrojo/community-timezone

Correct UTC time for meeting

castrojo committed:
More time zone fixes.

k8s-ci-robot committed:
Merge pull request #2162 from castrojo/meeting-fixes

More time zone fixes.

castrojo committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

Empower moderators to do the right thing

# This is the commit message #2:

Fix spelling mistake

jessfraz committed:
add SECURITY_CONTACTS

Signed-off-by: Jess Frazelle <acidburn@microsoft.com>

k8s-ci-robot committed:
Merge pull request #1934 from jessfraz/rawproc

add ProcMount option

webwurst committed:
Fix links to kustomize examples

wojtek-t committed:
Organize SLOs

castrojo committed:
First draft of a zoom moderation document

k8s-ci-robot committed:
Merge pull request #2164 from webwurst/patch-1

Fix links to kustomize examples

k8s-ci-robot committed:
Merge pull request #2146 from austbot/location-typo-apichange-docs

Fix incorrect path in api_changes doc

castrojo committed:
Add review feedback

Require host moderation.
Add section for editing video before publishing.
Grammar fixes.

k8s-ci-robot committed:
Merge pull request #2166 from castrojo/zoom-moderation

Initial Zoom moderation guidelines

k8s-ci-robot committed:
Merge pull request #2084 from k82cn/k8s_54313_doc

Doc about PID pressure taint.

k8s-ci-robot committed:
Merge pull request #2090 from wojtek-t/node_heartbeat_kep

KEP: efficient node heartbeats

AdamDang committed:
Typo fix: DeamonSet->DaemonSet

gyuho committed:
keps: fix typo in "0008-node-heartbeat.md"

Signed-off-by: Gyuho Lee <gyuhox@gmail.com>

k8s-ci-robot committed:
Merge pull request #2169 from AdamDang/patch-11

Typo fix: DeamonSet->DaemonSet

k8s-ci-robot committed:
Merge pull request #2170 from gyuho/typo

keps: fix typo in "0008-node-heartbeat.md"

hongchaodeng committed:
0008-node-heartbeat.md: fix typo

k8s-ci-robot committed:
Merge pull request #2172 from hongchaodeng/patch-1

0008-node-heartbeat.md: fix typo

wojtek-t committed:
Minot fixed to heartbeast KEP

k8s-ci-robot committed:
Merge pull request #2175 from wojtek-t/cleanup_node_heartbeat_kep

Minor fixes to heartbeat KEP

k8s-ci-robot committed:
Merge pull request #2168 from msau42/topology-design

Topology aware dynamic provisioning design (take 3)

k8s-ci-robot committed:
Merge pull request #2062 from jessfraz/security-owners

add SECURITY_CONTACTS to repo template

k8s-ci-robot committed:
Merge pull request #2179 from shubheksha/add-sig-storage-resources

Add Kubecon EU 2018 talks to contributing resources for sig-storage

cblecker committed:
Add SECURITY_CONTACTS for k/community

justinsb committed:
Fix name of linux foundation bot: thelinuxfoundation

linuxfoundation appears to be the org, thelinuxfoundation is the bot.

k8s-ci-robot committed:
Merge pull request #2184 from justinsb/linuxfoundation_to_thelinuxfoundation

Fix name of linux foundation bot: thelinuxfoundation

k8s-ci-robot committed:
Merge pull request #2185 from wojtek-t/cleanup_slos

Better organize performance SLOs - initial cleanup

k8s-ci-robot committed:
Merge pull request #2181 from cblecker/sec-community

Add SECURITY_CONTACTS for k/community

k8s-ci-robot committed:
Merge pull request #2158 from frapposelli/add-vmware-sig-subproject

Add SIG VMware ML and cloud-provider-vsphere subproject

Commit:
Add SIG VMware subproject call and meeting recordings

yujuhong committed:
Update the documentation of CRI log format

k8s-ci-robot committed:
Merge pull request #2187 from frapposelli/master

Add SIG VMware subproject call and meeting recordings

jdumars committed:
Changed company affiliation for jdumars

k8s-ci-robot committed:
Merge pull request #2190 from jdumars/sigupdates

Changed company affiliation for jdumars

k8s-ci-robot committed:
Merge pull request #2189 from yujuhong/cri-log-format

Update the documentation of CRI log format

wojtek-t committed:
Organize SLOs

codenrhoden committed:
Use singular form of DST

Daylight Saving Time (DST) is singular, not plural.

k8s-ci-robot committed:
Merge pull request #2193 from codenrhoden/dst

Use singular form of DST

Commit:
fixed link to owners in the maintainer section

k8s-ci-robot committed:
Merge pull request #2195 from teaguecole/fixownerlink

Fixed link to owners in the maintainer section

hh committed:
update apimachinery OWNER raw-github links

hh committed:
adding generated sig-api-machinery readme

Commit:
add poseidon to sig-scheduling subproject

wojtek-t committed:
Remove apimachinery SLIs

k8s-ci-robot committed:
Merge pull request #2197 from m1093782566/sig-scheduler-poseidon

Add poseidon to sig-scheduling subproject

k8s-ci-robot committed:
Merge pull request #2196 from hh/update-apimachinery-links

Update api-machinery OWNER links

k8s-ci-robot committed:
Merge pull request #2133 from AdamDang/patch-7

Typo fix: Kuberntes->Kubernetes

k8s-ci-robot committed:
Merge pull request #2198 from wojtek-t/organize_slos_2

Better organize performance SLIs/SLOs - part 2

k8s-ci-robot committed:
Merge pull request #1978 from rramkumar1/master

Add a retroactive KEP for IPVS
